File: leetcode/151. Reverse Words in a String/151.js
// https://leetcode.com/problems/reverse-words-in-a-string/
const reverseWords = (s) => s.split(" ").filter(e => e != "").reverse().join(" ");

// Tests
reverseWords("the sky is blue")   // "blue is sky the"
reverseWords("  hello world  ")   // "world hello"
reverseWords("a good   example")  // "example good a"

File: leetcode/151. Reverse Words in a String/151.hs
-- https://leetcode.com/problems/reverse-words-in-a-string/
import Data.List (intercalate)

reverseWords :: [Char] -> [Char]
reverseWords =  intercalate " " . reverse . words

-- Tests
main :: IO()
main = do
    print $ reverseWords "the sky is blue"      -- "blue is sky the"
    print $ reverseWords "  hello world  "      -- "world hello"
    print $ reverseWords "a good   example"     -- "example good a"
